core/templates/pages/exploration-editor-page/services/exploration-automatic-text-to-speech.service.spec.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the ExplorationAutomaticTextToSpeechService.
+ */
+
+// TODO(#7222): Remove the following block of unnnecessary imports once
+// the code corresponding to the spec is upgraded to Angular 8.
+import { UpgradedServices } from 'services/UpgradedServices';
+// ^^^ This block is to be removed.
+
+require('pages/exploration-editor-page/' +
+  'services/exploration-automatic-text-to-speech.service.ts');
+
+describe('Exploration Automatic Text To Speech Service', function() {
+  let eattss = null;
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    var ugs = new UpgradedServices();
+    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
```

Comment:
Wanted to check whether we need to provide all the upgraded services here? Can we only the required services here? [I'm concern about the time for running the unit test and wanted to check whether this can take extra time?]

Reply:
Hm, interesting point. Tagging @bansalnitish in case he has some suggestions.

Reply:
We are adding all the upgraded services in such unit tests and the reason for this is it could time consuming to go all the level up till the top to determine the which services are indirectly used by that particular service.

We are also migrating services parallelly so this piece of code goes away while the services are migrated.

Reply:
>  time consuming to go all the level up till the top to determine the which services are indirectly used by that particular service

@bansalnitish I'm expecting parent services to have imports for their dependencies, right? So why we need to import everything?

Reply:
Yes, you are 100% right parent services will import on their own what ever they need. The thing is we have a downgradeModule at the end of each Angular file which works only when it sees first Angular component and basically it does not work while the frontend tests run. 

So, if you have some service A which is in AngularJS which imports some service B which is in AngularJS and service B imports some service C which is in Angular, B cannot import C while tests run and so there would be a problem to service A as well. The reason for this is service C is not downgraded to Angular and hence cannot be used in AngularJS code. 

There could be so many services like C that are indirectly used by A and we need to provide an instance of all such services which is what this piece of code does.

Does it makes sense now?